### DISCUSSION TITLE
add validation/pattern for DatagridStringFilter

### DISCUSSION THREAD

**Discussion by KingMario** [vmware-clarity/ng-clarity#474] [Ideas]
Sometimes we need to limit the user input for string filter.
For example, for a server-driven filter of numeric field, we want to use the string filter accepts only digits input.

Please consider adding an optional validation or pattern property for the string filter that it'll emit changes if only the user input is valid.

Thanks.

**Comment by Jinnie** [comment]
Jeff is working on a global initiative for our future filters.
cc-ing, so he can capture this request: @lerch015

**Comment by KingMario** [comment]
Hi @Jinnie,
Hi @lerch015

A third-party cleave.js library can be used for configuring input restriction.

And I've created a directive to use it directly: https://www.npmjs.com/package/ngx-cleave-directive. Please take a look to see whether it can be used here.

Of course, simply using angualr built-in validation or so can do the job too.
